cmake/EthCompilerSettings.cmake

```diff
@@ -157,9 +157,10 @@ elseif (DEFINED MSVC)
 	add_compile_options(/wd4800)					# disable forcing value to bool 'true' or 'false' (performance warning) (4800)
 	add_compile_options(-D_WIN32_WINNT=0x0600)		# declare Windows Vista API requirement
 	add_compile_options(-DNOMINMAX)					# undefine windows.h MAX && MIN macros cause it cause conflicts with std::min && std::max functions
-	add_compile_options(/utf-8)					# enable utf-8 encoding (solves warning 4819)
+	add_compile_options(/utf-8)						# enable utf-8 encoding (solves warning 4819)
 	add_compile_options(-DBOOST_REGEX_NO_LIB)		# disable automatic boost::regex library selection
 	add_compile_options(-D_REGEX_MAX_STACK_COUNT=200000L)	# increase std::regex recursion depth limit
+	add_compile_options(/permissive-)				# specify standards conformance mode to the compiler
```

Comment:
It is actually funny what the documentation has to say about that flag:

`"Use this option to help you identify and fix conformance issues in your code"`

But reading through the whole documentation at:

* https://docs.microsoft.com/en-us/cpp/build/reference/permissive-standards-conformance?view=vs-2019

makes me wonder whether this option is the right way to address the ADL problem.

The option `/permissive-` is enabling/disabling a few options that all don't really look like they relate to the actual thing.

- IDE will highlight non-standard-conforming code fragments
- sets `/Zc:referenceBinding`
- sets `/Zc:strictStrings`
- sets `/Zc:rvalueCast`
- sets `/Zc:ternary`
- sets `/Zc:twophase-` (with trailing dash)

The documentation for the last option behind:

* https://docs.microsoft.com/en-us/cpp/build/reference/zc-twophase?view=vs-2019

also states that `"When /permissive- isn't specified, the non-conforming behavior is the default."`, so this is where we might have been running in. I think it might be better to try this option then.

Reply:
> whether this option is the right way to address the ADL problem

I believe "Hidden friend name lookup rules" section illustrates issues similar to our one.